SSA/Projects/Scf/Scf.lean

```diff
@@ -51,52 +57,138 @@ instance : OpSignature Op Ty where
 
 #check uncurry
 
+
+/-# define what it means for a loop body (a function of type `(i : Int) × (v : t) → t` to be index invariant -/
+
+/-- A loop body receives the current value of the loop induction variable, and the current loop carried value.
+The loop body produces the value of this loop iteration.
+
+Consider the loop: `for(int i = 0; i >= -10; i -= 2)`. In this case, the value `i` will be the sequence
+of values `[0, -2, -4, -6, -8, -10]`. The value `i` is the loop induction variable.
+
+This value is distinct from the *trip count*, which is the number of times the loop body has been executed so far, which is `[0, 1, 2, 3, 4]`.
+
+The `LoopBody` does *not* provide access to the trip count, only to the loop induction variable.
+-/
+abbrev LoopBody (t : Type) : Type := Int → t → t
+
+namespace LoopBody
+
 /-- Convert a function `f` which is a single loop iteration into a function
   that iterates and updates the loop counter. -/
-def loop_counter_decorator (δ: Int) (f : Int → α → α) : Int × α → Int × α :=
+def CounterDecorator (δ: Int) (f : LoopBody α) : Int × α → Int × α :=
   fun (i, v) => (i + δ, f i v)
```

Comment:
This is should be `lowerCamelCase`, since it's a function returning `Int \times \a : Type _`

Reply:
Done.